base/show.jl

```diff
@@ -1495,7 +1495,8 @@ end
 function array_eltype_show_how(X)
     e = eltype(X)
     leaf = isleaftype(e)
-    plain = e<:Number || e<:AbstractString
+    plain = e<:Number || e<:AbstractString ||
```

Comment:
This code (both the original and the modified versions) sounds too ad-hoc to me. Shouldn't we export a method to allow package authors to decide what's the most appropriate behavior for their type?